migrate/migrate_service.go

```diff
@@ -59,8 +59,28 @@ func (s *serviceMigrate) Up() (err error) {
 	return
 }
 
-func (s *serviceMigrate) Down() error {
-	return errors.New("not implemented")
+func (s *serviceMigrate) Down() (err error) {
+	current, errSchema := s.CurrentSchemaVersion()
+	latest := s.LatestSchemaVersion()
+	if errSchema != nil {
+		return errSchema
+	}
+
+	index := current
+	for index >= 0 {
+		s.logger.Info(fmt.Sprintf("applying migration %d of %d", index, latest))
+		err = s.migrations[index-1].Down()
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("error on migration down %d of %d", index, latest), zap.Error(err))
+			break
+		}
+		index--
+	}
```

Comment:
Shouldn't this loop start at the latest version and go down until the current one (or vice versa)? Because currently this command will undo all the migrations.
I am not sure, but at least the logs seem strange to me, because they start at the current version, and are compared to the latest (btw, the latest variable is just used for it), and it would be going backwards like this:
```
applying migration 3 of 5
applying migration 2 of 5
applying migration 1 of 5
applying migration 0 of 5
```

Reply:
nope, the migration should always consider the current version and then do something. In this case rolling back all migrations. It is also common to see some migrations tools to support the specific version you want to migrate up or down, but in our scenario the migration svc was coded too simple to support that.

btw the message should be `rolling back migration X of Y` instead of `applying migration X of Y`

Reply:
Right, I will keep this comment to fix the logs.

Reply:
I am reworking this to a stack, so we can fill the stack on migrate up and pop of the stack on migrate down.